c7n/resources/iam.py

```diff
@@ -817,6 +817,44 @@ def process(self, resources):
                     pass
 
 
+@Role.action_registry.register('delete')
+class RoleDelete(BaseAction):
+    """Delete an IAM Role.
+
+    For example, if you want to automatically delete an unused IAM role.
+
+    :example:
+
+      .. code-block:: yaml
+
+        - name: iam-delete-unused-role
+          resource: iam-role
+          filters:
+            - type: usage
+              match-operator: all
+              LastAuthenticated: null
+          actions:
+            - delete
+
+    """
+    schema = type_schema('delete')
+    permissions = ('iam:DeleteRole',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('iam')
+        for r in resources:
+            try:
+                client.delete_role(RoleName=r['RoleName'])
+            except ClientError as e:
+                if e.response['Error']['Code'] == 'DeleteConflict':
```

Comment:
use error class from client.exceptions.DeleteConflictException and NoSuchEntityException.. thats preferred over string matching on error codes.

Reply:
Should we log in case of conflict?

Reply:
```
error = None
for r in resources:
   try:
       do_something
   except NotFound:
        continue
   except OtherErrorContinue as e:
        error = e
if error:
   raise error
```

Reply:
a long for delete conflict makes sense

Reply:
Ah, ok! Understood! Thanks! I over thought it! I thought if there are multiple delete conflicts, we might have to capture all and raise a list.